### ISSUE TITLE
AttributeError: 'EnvisalinkAlarmPanel' object has no attribute 'self'

### ISSUE THREAD

**Issue by rct** [ufodone/envisalink_new#20]
Testing disconnection with 0.5.0, got these two exceptions. the first might be expected though the second one isn't.

Envisalink appears NOT recover/reconnect after this. Home Assistant needs to be restarted to resolve. 

```
2023-02-12 13:16:31.516 DEBUG (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Waiting for data from EVL
2023-02-12 13:16:36.521 DEBUG (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Waiting for data from EVL
2023-02-12 13:16:39.217 ERROR (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Caught unexpected exception: IncompleteReadError('0 bytes read on a total of undefined expected bytes')
2023-02-12 13:16:39.222 DEBUG (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Cleaning up from disconnection with server.
2023-02-12 13:16:39.224 DEBUG (MainThread) [custom_components.envisalink_new] state_updated for 'Home Keypad'
2023-02-12 13:16:39.224 DEBUG (MainThread) [custom_components.envisalink_new] state_updated for 'Home'
2023-02-12 13:16:39.224 DEBUG (MainThread) [custom_components.envisalink_new] state_updated for 'Garden Level Entrance'

[...SNIP list of zone state updates...]

2023-02-12 13:16:39.226 DEBUG (MainThread) [custom_components.envisalink_new] state_updated for 'Basement Bathroom Flood Sensor'
2023-02-12 13:16:39.226 ERROR (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Reconnection attempt in 30s
2023-02-12 13:17:09.228 DEBUG (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Starting read loop.
2023-02-12 13:17:09.228 INFO (MainThread) [custom_components.envisalink_new.pyenvisalink.envisalink_base_client] Started to connect to Envisalink... at AlarmServerHost:4025
2023-02-12 13:17:09.298 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py", line 186, in connect
    self._reader, self._writer = await asyncio.wait_for(
  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.10/asyncio/streams.py", line 47, in open_connection
    transport, _ = await loop.create_connection(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect
    return await fut
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.201.11', 4025)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py", line 126, in read_loop
    await self.connect()
  File "/config/custom_components/envisalink_new/pyenvisalink/envisalink_base_client.py", line 204, in connect
    _LOGGER.error("Unable to connect to envisalink at %s: %r", self._alarmPanel.self, ex)
AttributeError: 'EnvisalinkAlarmPanel' object has no attribute 'self'

```

**Comment by ufodone** [comment]
These should be fixed in 0.5.1 now.

**Comment by rct** [comment]
0.5.1 handles the disconnect/reconnect gracefully. 

Also successful, it detects and disconnects if the EVL stops sending responses, but doesn't close the TCP connection (simulated by SIGSTOP AlarmServer).

However, I noticed that when pyenvisalink reconnected to the stopped process, it didn't time out waiting for the initial 5053.  I only waited for 3 minutes before letting the stopped process resume. I probably should have waited longer.

This test should somewhat simulate the case where the Envisalink device has enough of a TCP stack still running to accept connections, but not to ever respond at the application layer.  (On a *nix type host this is the equivalent of the kernel is fine for processing TCP, but the user level process is alive but broken/zombied.)
